python/ray/data/dataset.py

```diff
@@ -921,6 +943,11 @@ def filter(
                 fixed-sized worker pool of size ``n``, specify ``concurrency=n``.
                 For an autoscaling worker pool from ``m`` to ``n`` workers, specify
                 ``concurrency=(m, n)``.
+            ray_remote_args_fn: A function that returns a dictionary of remote args
+                passed to each map worker. This function will be called each time prior
+                to initializing the worker. Args returned from this dict will always
+                override the args in ``ray_remote_args``. Note: this is an advanced,
+                experimental feature.
```

Comment:
nit, maybe add a note that the purpose of this argument is to allow generating dynamic arguments for each actor/task